metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -99,11 +102,19 @@ def report_warning(self, key: str, reason: str) -> None:
         warnings.append(reason)
         self.warnings[key] = warnings
 
+    def warning(self, key: str, reason: str) -> None:
+        self.report_warning(key, reason)
```

Comment:
Should report_warning and report_failure eventually be private methods, or do you see a case for putting a warning / failure in the report without logging it?

Reply:
Yup that's the goal - I want to get rid of direct calls to `report_warning` and `report_failure`. A bunch of sources also have helper methods that are basically copies of this warning method, and I want to get rid of those too

https://github.com/datahub-project/datahub/blob/bf9f380350db5967634638f089c78a516dfa2b6a/metadata-ingestion/src/datahub/ingestion/source/redash.py#L364